dev-dotnet/dotnet-sdk/dotnet-sdk-5.0.203.ebuild

```
@@ -0,0 +1,224 @@
+	add_patch runtime "${FILESDIR}/dotnet-sdk-5.0.203-runtime-locate-ranlib.patch"
+
+	default
+}
+
+src_configure() {
+	dotnet_find
+
+	# SDK is modified in-place...
+	if use system-bootstrap; then
+		mkdir -p dotnet
+		ln -s "${DOTNET_ROOT}"/{host,packs,shared} dotnet || die
+		cp -a "${DOTNET_ROOT}"/{dotnet,sdk} dotnet || die
+	else
+		ln -s "${DOTNET_ROOT}" dotnet || die
+	fi
+}
+
+src_compile() {
+	dotnet_find
+	dotnet_find_reference
+	rm -rf packages/archive  # Required for --with-packages to work
+	local mybuildargs=(
+		--with-sdk dotnet
+		--with-packages "${DOTNET_ROOT}/source-artifacts"
+		--with-ref-packages "${DOTNET_REFERENCE}"
+		--
+
+		/p:UseSystemLibraries=true
+		/p:UseSystemLibunwind=$(usex system-libunwind true false)
+
+		/p:LogVerbosity=normal
+		/p:MinimalConsoleLogOutput=false
+		/verbosity:normal
+
+		# Not sure if this has any effect on the subprocesses
+		/maxCpuCount:$(makeopts_jobs)
+
+		# /p:AdditionalBuildArguments are passed to src/runtime.*/eng/build.sh
+		# /p:NativeBuildArguments are passed to src/runtime.*/eng/native/build-commons.sh
+		/p:AdditionalBuildArguments="-cmakeargs -DCMAKE_VERBOSE_MAKEFILE=ON"
+		/p:NativeBuildArguments="${CMAKE_GENERATOR_FLAG} -numproc $(makeopts_jobs)"
+		# If any dotnet maintainer is reading this: Please honor the
+		#  CMAKE_MAKEFILE_GENERATOR and MAKEFLAGS variables if set.
+	)
+
+	# Adapted from www-client/firefox
+	if ${FORCE_CLANG} && ! tc-is-clang; then
+		einfo "Enforcing the use of Clang due to build errors..."
+		AR=llvm-ar
+		CC=${CHOST}-clang
+		CXX=${CHOST}-clang++
+		NM=llvm-nm
+		OBJCOPY=llvm-objcopy
+		OBJDUMP=llvm-objdump
+		RANLIB=llvm-ranlib
+		if is-flagq '-flto*' && ! has_version 'sys-devel/clang[default-lld]' && ! is-ldflagq '-fuse-ld=lld'; then
+			has_version sys-devel/lld || die "Please install sys-devel/lld (or sys-devel/clang[default-lld]) to build with LTO"
+			einfo "Enforcing the use of LLD for Clang LTO..."
+			append-ldflags '-fuse-ld=lld'
+		fi
+		strip-unsupported-flags
+	fi
+
+	# For the CLR_* variables, the full path is necessary, see:
+	# - src/runtime.*/eng/native/init-compiler.sh
+	# - src/runtime.*/eng/native/configuretools.cmake
+	# NM, OBJCOPY and OBJDUMP don't seem to be used currently (ver 5.0.203).
+	for var in AR CC CXX NM OBJCOPY OBJDUMP RANLIB; do
+		local prog="$(tc-get${var})"
+		local path="$(type -p "${prog}")"
+		test -n "${path}" || die "Can't find ${prog} (\$${var}) in \$PATH!"
+		export CLR_${var}="${path}"
+	done
+
+	# EXTRA_* currently (ver 5.0.203) don't seem to be required, but might be
+	#  in the future. Specifying flags twice never hurt anyone, anyway.
+	# See src/runtime.*/eng/native/build-commons.sh
+	export EXTRA_CFLAGS="${CFLAGS}"
+	export EXTRA_CXXFLAGS="${CXXFLAGS}"
+	export EXTRA_LDFLAGS="${LDFLAGS}"
+
+	./build.sh "${mybuildargs[@]}" || die
+}
+
+src_install() {
+	dodir /usr/lib/dotnet
+	tar xf artifacts/*/Release/dotnet-sdk-*.tar.gz -C "${ED}/usr/lib/dotnet"
```

Always use || die on external commands:
```suggestion
	tar xf artifacts/*/Release/dotnet-sdk-*.tar.gz -C "${ED}"/usr/lib/dotnet || die
```